### DISCUSSION TITLE
[V5] Model uuid primary key set to 0

### DISCUSSION THREAD

**Discussion by unkobweb** [adonisjs/core#3037] [Help] [answered]
Hi, i use stripe and a mysql db to store my customer data (just his id in my app and the stripe customer_id) so there is my code : 
```typescript
async getCustomer({request, response}: HttpContextContract) {
    try {

        const user = request.input('user')
        const customer = await CustomerUser.find(user.id)

        if (!customer) {
            const stripe_customer = await stripe.customers.create({
                description: user.id,
                email: user.email
            })

            const klasseur_customer = new CustomerUser()
            
            console.log(user.id) // The console return me the UUID of the suer without problems
            await klasseur_customer.fill({id: user.id, customer_id: stripe_customer.id}).save()           
            
            response.send(klasseur_customer) // Line who return the problem
        } else {
            response.send(customer) // Give me back the good object
        }
    } catch (error) {
        console.log(error)
        response.badRequest(error.messages || "Le client Stripe n'a pas pu être créé/trouvé")
    }
}
```

But where I have the `response.send(klasseur_customer)` normally the serveur have to return me an object like this :
```JSON
{
  "id": "f1e1320a-d6fb-4046-a8f1-4cca3165ab80",
  "customer_id": "cus_K1qCbsUYcNrsUM",
  "createdAt": "...",
  "updatedAt": "..."
}
```

But for an unknowned reason the server send me back this : 
```JSON
{
  "id": 0,
  "customer_id": "cus_K1qCbsUYcNrsUM",
  "createdAt": "...",
  "updatedAt": "..."
}
```

And if i do the same request just after, this time my server found an entry in my db and give my back the good object who is :
```JSON
{
  "id": "f1e1320a-d6fb-4046-a8f1-4cca3165ab80",
  "customer_id": "cus_K1qCbsUYcNrsUM",
  "createdAt": "...",
  "updatedAt": "..."
}
```
I tried with the create method, with just set every value of every key of my object before doing a save() or with this fill() method, all of them give me the same problem.

Here is my migration file: 
```typescript
import BaseSchema from '@ioc:Adonis/Lucid/Schema'

export default class CustomerUsers extends BaseSchema {
  protected tableName = 'customer_users'

  public async up () {
    this.schema.createTable(this.tableName, (table) => {
      table.uuid('id').primary()
      table.string('customer_id', 255).notNullable()
      table.timestamps(true, true)
    })
  }

  public async down () {
    this.schema.dropTable(this.tableName)
  }
}
```
And here is my model : 
```typescript
import { DateTime } from 'luxon'
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'

export default class CustomerUser extends BaseModel {
  @column({ isPrimary: true })
  public id: string

  @column()
  public customer_id: string

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime
}
```

Is anyone have an idea of what am I doing wrong ?
Thanks

**Comment by binsarjr** [comment]
you can try this
https://docs.adonisjs.com/reference/orm/base-model#refresh

or using hook
```javascript
export default class CustomerUser extends BaseModel {
  // ...
  @beforeCreate()
  public static async createUUID(model: CustomerUser) {
    model.id = uuid() // uuid() is a library for creating uuid format
  }
}
```

this happens because the uuid is created by mysql and is not readable by the current program code so it needs to refresh the data first. But, you can outsmart it by entering the uuid from the program code

**Comment by RomainLanz** [comment, marked as answer]
Hey @unkobweb! 👋 

MySQL does not work with UUID.
You should use PostgreSQL to have a method that generates UUID for you.

If you would like to keep MySQL, you need to generate the UUID yourself using the `beforeCreate()` hook and the library you want to use.

```ts
import { randomUUID } from 'crypto'
// ...

class CustomerUser extends BaseModel {
  // ...

  @beforeCreate()
  public static computeUuid(model: CustomerUser) {
    model.id = randomUUID()
  }

  // ...
}
```

📚 https://docs.adonisjs.com/reference/orm/decorators#beforecreate

**Reply by unkobweb** [reply]
Thanks for your reply but I don't want to generate this UUID but just set the UUID of the user (`user.id` in my code) so there is no UUID generation in my code.

**Reply by RomainLanz** [reply]
I misread your question.

Since you are assigning yourself the ID, you need to change `selfAssignPrimaryKey`.

```ts
class User extends BaseModel {
  public static selfAssignPrimaryKey = true

  @column({ isPrimary: true })
  public userId: string
}
```

📚 https://docs.adonisjs.com/reference/orm/base-model#static-selfassignprimarykey

**Reply by unkobweb** [reply]
It's work perfectly 👌
